### ISSUE TITLE
Anna_KaRNNa variable sharing error

### ISSUE THREAD

**Issue by llealgt** [udacity/deep-learning#132]
Im getting :Trying to share variable rnn/multi_rnn_cell/cell_0/basic_lstm_cell/weights, but specified shape (1024, 2048) and found shape (595, 2048).

To verify whats wrong in mi code, i run the solutions notebook with no modification and it gets the same error.

**Comment by Fiasco-zh** [comment]
I'm in same error in tensorflow 1.2 while there will be another error in tensorflow 1.1.

**Comment by KylixC** [comment]
Back your tensorflow to 1.0.0, can solve this error.

**Comment by lizhengwei1992** [comment]
I have the same error 
"Trying to share variable rnn/multi_rnn_cell/cell_0/basic_rnn_cell/kernel, but specified shape (1024, 2048) and found shape (513, 2048)."

**Comment by lizhengwei1992** [comment]
@llealgt 
I success like this:

```
stacked_rnn = []
for layer in range(3):
    stacked_rnn.append(tf.contrib.rnn.BasicRNNCell(n_hidden))
rnn_cells= tf.contrib.rnn.MultiRNNCell(cells=stacked_rnn, state_is_tuple=True)

 outs, states = tf.contrib.rnn.static_rnn(rnn.cells, inputs, dytper=tf.float32)
```

**Comment by taylordelehanty** [comment]
@lizhengwei1992, Yep, I did something similar:

```
    # Build the RNN layers
    # The next few lines seems to be where the problem is, according to:
    # https://stackoverflow.com/questions/44615147/valueerror-trying-to-share-variable-rnn-multi-rnn-cell-cell-0-basic-lstm-cell-k
    
    def lstm_cell():
        return tf.contrib.rnn.BasicLSTMCell(lstm_size)
    # Original:
    # drop = tf.contrib.rnn.DropoutWrapper(lstm, output_keep_prob=keep_prob)
    # cell = tf.contrib.rnn.MultiRNNCell([drop] * num_layers)
    cell = tf.contrib.rnn.MultiRNNCell([lstm_cell() for i in range(num_layers)])
```

**Comment by arthurcgusmao** [comment]
@taylordelehanty Thanks, that worked for me as well, but you forgot the dropout:

```python
        def lstm_cell():
            lstm = tf.contrib.rnn.BasicLSTMCell(lstm_size)
            drop = tf.contrib.rnn.DropoutWrapper(lstm, output_keep_prob=keep_prob)
            return drop
        cell = tf.contrib.rnn.MultiRNNCell([lstm_cell() for i in range(num_layers)])
```

**Comment by zh794390558** [comment]
@arthurcgusmao Thx

**Comment by abhiojha8** [comment]
There are no issues with Tensorflow 1.0.0. The newer versions of TF will cause this error.
